Neos.Flow/Classes/Security/Policy/RoleIdentifiers.php

```diff
@@ -0,0 +1,116 @@
+<?php
+declare(strict_types=1);
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+namespace Neos\Flow\Security\Policy;
+
+use Neos\Flow\Annotations as Flow;
+
+/**
+ * A collection class containing all roles for an account
+ *
+ * @Flow\Proxy(false)
+ */
+final class RoleIdentifiers implements \JsonSerializable, \IteratorAggregate, \Countable
+{
+
+    /**
+     * @var string[]
+     */
+    private $roleIdentifiers;
+
+    private function __construct(array $roleIdentifiers)
+    {
+        $this->roleIdentifiers = $roleIdentifiers;
+    }
+
+    /**
+     * @return self
+     */
+    public static function create(): self
+    {
+        return new static([]);
+    }
+
+    /**
+     * @param string[] $roleIdentifiers
+     * @return static
+     */
+    public static function fromArray(array $roleIdentifiers): self
+    {
+        $processedRoleIdentifiers = [];
+        array_walk($roleIdentifiers, static function ($roleIdentifier) use (&$processedRoleIdentifiers) {
+            $processedRoleIdentifiers[(string)$roleIdentifier] = $roleIdentifier;
+        });
+        return new static($processedRoleIdentifiers);
```

Comment:
I would suggest to turn this into

```suggestion
        foreach ($roleIdentifiers as $roleIdentifier) {
            if (!is_string($roleIdentifier)) {
                throw new \InvalidArgumentException(sprintf('Expected an array of strings, got: %s', gettype($roleIdentifier)), 1602174016);
            }
            $processedRoleIdentifiers[$roleIdentifier] = $roleIdentifier;
        }
        return new static($processedRoleIdentifiers);
```

..easier to read (IMO) and type safe